ledger/src/blockstore.rs

```diff
@@ -1442,7 +1451,16 @@ impl Blockstore {
     }
 
     pub fn get_data_shred(&self, slot: Slot, index: u64) -> Result<Option<Vec<u8>>> {
-        self.data_shred_cf.get_bytes((slot, index))
+        use crate::shred::SHRED_PAYLOAD_SIZE;
+        self.data_shred_cf.get_bytes((slot, index)).map(|data| {
+            data.map(|mut d| {
+                // Only data_header.size bytes stored in the blockstore so
+                // pad the payload out to SHRED_PAYLOAD_SIZE so that the
+                // erasure recovery works properly.
+                d.resize(cmp::max(d.len(), SHRED_PAYLOAD_SIZE), 0);
```

Comment:
should there also be a check on insert that the size doesn't exceed `SHRED_PAYLOAD_SIZE`?

Reply:
Yeah, I could see that being a reasonable check to make. Although, I think we'd actually want to compare to `SIZE_OF_DATA_SHRED_PAYLOAD` on insert; because we know we there shouldn't be anything in the "restricted" area at the end

Reply:
Oops, CI helped me see you were right :). `SIZE_OF_DATA_SHRED_PAYLOAD` excludes the headers, confused myself there a bit ... thank goodness for unit tests